BASE/src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -595,48 +689,6 @@ public void TrackDependency(DependencyTelemetry telemetry)
             }
         }
 
-        /// <summary>
-        /// Send information about availability of an application.
-        /// </summary>
-        /// <param name="name">Availability test name.</param>
-        /// <param name="timeStamp">The time when the availability was captured.</param>
-        /// <param name="duration">The time taken for the availability test to run.</param>
-        /// <param name="runLocation">Name of the location the availability test was run from.</param>
-        /// <param name="success">True if the availability test ran successfully.</param>
-        /// <param name="message">Error message on availability test run failure.</param>
-        /// <param name="properties">Named string values you can use to classify and search for this availability telemetry.</param>
-        /// <remarks>
-        /// <a href="https://go.microsoft.com/fwlink/?linkid=517889">Learn more</a>
-        /// </remarks>
-        public void TrackAvailability(string name, DateTimeOffset timeStamp, TimeSpan duration, string runLocation, bool success, string message = null, IDictionary<string, string> properties = null)
-        {
-            var availabilityTelemetry = new AvailabilityTelemetry(name, timeStamp, duration, runLocation, success, message);
-
-            if (properties != null && properties.Count > 0)
-            {
-                Utils.CopyDictionary(properties, availabilityTelemetry.Properties);
```

Comment:
The old version used [`Utils.CopyDictionary`](https://github.com/microsoft/ApplicationInsights-dotnet/blob/98bc6c540e2dcccc78e4b356cd70e03d146a01ad/BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Utils.cs#L25), which does empty string and duplicate key checks.

Should the same pattern apply to new code?